### ISSUE TITLE
Flood on roads with an aggregation layer fails

### ISSUE THREAD

**Issue by Gustry** [inasafe/inasafe#2750]
### Problem
I'm trying to run Flood on Roads with an aggregation layer but the IF raises an error.

![selection_073](https://cloud.githubusercontent.com/assets/1609292/14907778/bdc17e56-0df7-11e6-8a56-3fdbf288241b.png)

```
Problem
The following problem(s) were encountered whilst running the analysis.
KeyError

Détails
These additional details were reported when the problem occurred.

An exception occurred when running the impact analysis.
Jalan Cideng Barat Aggregation error occurred

Diagnostics
In file "/home/etienne/.qgis2/python/plugins/inasafe/safe/impact_functions/base.py", line 702, in run_analysis self._run_aggregator()
In file "/home/etienne/.qgis2/python/plugins/inasafe/safe/impact_functions/base.py", line 1359, in _run_aggregator self.aggregator.aggregate(self.impact)
In file "/home/etienne/.qgis2/python/plugins/inasafe/safe/impact_statistics/aggregator.py", line 589, in aggregate qgis_impact_layer, safe_impact_layer)
In file "/home/etienne/.qgis2/python/plugins/inasafe/safe/impact_statistics/aggregator.py", line 673, in _aggregrate_vector_impact self._aggregate_line_impact(safe_impact_layer)
In file "/home/etienne/.qgis2/python/plugins/inasafe/safe/impact_statistics/aggregator.py", line 1026, in _aggregate_line_impact polygon_index = agg_attribute_dict[polygon_name]
```

### Environment
* Flood polygon, Roads and the Grid are in the test data folder from develop
* InaSAFE develop

**Comment by Gustry** [comment]
I think that this bug happens if your exposure layer AND your aggregation layer use the same name field in the attribute table. In this case, the field "name" is used in both layers.

**Comment by Gustry** [comment]
![selection_008](https://cloud.githubusercontent.com/assets/1609292/16619410/3823f306-438e-11e6-87c8-aaebdf794d89.png)

**Comment by NyakudyaA** [comment]
yes @Gustry  I have tested with the same dataset and have an error but when I rename the column from the aggregation layer  to another name then it is happy.

**Comment by ismailsunni** [comment]
Hi @Gustry  @NyakudyaA I made a fix in #3128
